src/com/google/javascript/jscomp/NodeTraversal.java

```diff
@@ -657,6 +659,41 @@ private void traverseFunction(Node n, Node parent) {
     popScope();
   }
 
+  /** Traverses a class. */
+  private void traverseClass(Node n, Node parent) {
+    Preconditions.checkState(n.getChildCount() == 3, n);
+    Preconditions.checkState(n.isClass());
+
+    final Node className = n.getFirstChild();
+    boolean isClassExpression = (parent != null)
```

Comment:
How does this occur? (CLASS node with null parent)

I'd think that would be impossible in a full AST